transport/udp.go

```diff
@@ -27,52 +26,41 @@ func ListenUDP(addr string) (*UDPListener, error) {
 
 	return &UDPListener{
 		addr: addr,
-		pool: sync.Pool{
-			New: func() interface{} {
-				return make([]byte, 10*1024)
-			},
-		},
 		conn: conn,
 	}, nil
 }
 
 // UDPListener handles reading packets from the underlying UDP connection.
 type UDPListener struct {
 	addr string
-	pool sync.Pool
 	conn *net.UDPConn
 }
 
 // Next returns the next packet from UDP and it's associated source address. Should an error occur, then it is returned.
 // A source address may be returned alongside an error for further reporting in the event of abuse/debugging.
 func (u *UDPListener) Next() (packet *pb.Packet, source *net.UDPAddr, err error) {
-	buf := u.pool.Get().([]byte)
-	defer func() {
-		if buf != nil {
-			u.pool.Put(buf)
-		}
-	}()
-
-	n, source, err := u.conn.ReadFromUDP(buf)
+	var buf [10 * 1024]byte
```

Comment:
Was there a benefit of having this buffer pooled?

Reply:
it seemed more useful in an earlier refactor. i agree this is better

Reply:
Note, if it's still necessary, we can use a single cached buffer for the udpclient rather than use a pool.